### ISSUE TITLE
An ill-conditioned pixel costs its whole record: no CRLB EVD fallback, and the validity mask discards good displacement

### ISSUE THREAD

**Issue by mortonanalytics** [morton-analytics-llc/dolphinRust#34]
Two independent hardening items found while diagnosing #29. Neither is the SHP fix; both
limit the blast radius of the *next* ill-conditioned pixel.

## 1. `crlb.rs` has no EVD fallback

`estimator.rs::process_coherence_matrix` falls back to EVD when `Γ` is singular — the
comment calls it "dolphin's NaN-triggered fallback" and `dolphin-phaselink/CLAUDE.md` says
to keep it. `crlb.rs::crlb_pixel` has no equivalent: a non-PD `Γ` returns NaN on every
non-reference date, terminally.

So the linked phase survives an indefinite `Γ` but the bound does not, and the bound is what
the validity mask keys on. After #29 this still affects **0.69%** of the MMX1/ICMX frame.

Options: mirror the EVD fallback; or emit a bound derived from the regularized `Γ` and flag
it; or return a sentinel the mask can treat as "no bound" rather than "no data". Whichever is
chosen needs an oracle check against dolphin v0.42's actual singular-matrix behaviour, which
our pinned v0.35.0 oracle cannot answer — see the open re-pin question in #25.

## 2. `apply_validity_mask` discards displacement when only the *bound* is unusable

`displacement.rs::apply_validity_mask` (~line 579) gates `validity_mask` on
`velocity.is_finite()` and then NaNs **every** layer: displacement, velocity, temporal
coherence, CRLB, posterior variance, residual.

Because `fit_velocity` uses CRLB-derived precisions whenever `use_coherence_weights` **or**
`write_velocity_uncertainty` is set, one NaN σ zeroes the weights, NaNs the velocity, and
erases otherwise good displacement. That is how a bound-only failure became a data-loss
failure at ICMX — and why clearing `use_coherence_weights` did not avoid it.

A pixel with unusable uncertainty may still have perfectly good displacement. Proposal:
separate "no bound" from "no data" so uncertainty layers go NaN while displacement and
temporal coherence survive, with the validity mask distinguishing the two.

## Why together

Either alone reduces the damage; both together mean an ill-conditioned pixel costs its
uncertainty estimate and nothing else. Worth one design pass rather than two patches.

## Note

Raising `beta` is **not** a substitute. At ICMX, `beta: 0.01` still leaves `Γ`'s smallest
eigenvalue at −0.070; it needs roughly `β > 0.08`, which is a real change to the estimator
that would need its own parity work.

**Comment by mortonanalytics** [comment]
**Correction to part 1 — withdrawn.** I checked the v0.42 forward oracle (`oracle/.venv-v042`,
which exists, so this was never blocked on #25 as the issue claimed).

dolphin v0.42 `phase_link/crlb.py` deliberately NaNs singular blocks:

```
221:  Set output to nan where |Γ| is (near) zero.
248:  # overwrite sigma on zero blocks to NaN to mimic NumPy error/NaN
250:  nanv = jnp.full(sig.shape, jnp.nan, dtype=sig.dtype)
```

and `oracle/fixtures/crlb_singular_sigma_v042.npy` is exactly that: `[0, nan, nan, ...]`
per pixel, already pinned green by `quality_v042_contract`. **Our NaN is correct parity.**
Adding an EVD fallback to `crlb.rs` would break it. Part 1 is withdrawn.

**Part 2 is the whole bug, and it is bigger than the mask.** A NaN bound reaches the data
path twice, not once:

1. `interferogram_precisions` / `date_precisions` map non-finite σ to weight **0**. A pixel
   whose CRLB is NaN therefore gets an all-zero weight vector, the normal equations go
   singular, and *displacement itself* comes out NaN — before any masking.
2. `apply_validity_mask` then gates `validity_mask` on `velocity.is_finite()` and blanks
   every layer.

So the scope is: **a missing weight is missing information, not evidence the data is bad.**
Proposed fix — per pixel, if any required precision is non-finite, fall back to *uniform*
weights for that pixel in both the displacement inversion and the velocity fit, and keep the
uncertainty outputs (`crlb_sigma`, `velocity_sigma`, posterior variance) NaN to mark "no
bound available".

Worth noting why that fallback is nearly free: for a single-reference interferogram network
the SBAS system is exactly determined, so weights cancel and the fallback displacement is
*identical* to the weighted one (measured this way on the MMX1/ICMX frame — weighted vs
unweighted max|diff| = 0). It is only a real estimator change for over-determined networks
(`max_bandwidth` set), which should be documented rather than hidden.

**Comment by mortonanalytics** [comment]
Fixed in `37b3937` (on `main`, CI green).

**Part 1 stayed withdrawn.** dolphin v0.42 NaNs a singular block deliberately and
`quality_v042_contract` already pins that against
`oracle/fixtures/crlb_singular_sigma_v042.npy`. Our NaN bound was correct parity all along;
adding an EVD fallback would have broken it.

**Part 2 was the whole bug, and it reached the data twice.** The NaN arrived at
`interferogram_precisions` / `date_precisions` as a **zero weight**, making the normal
equations singular and destroying the displacement *before* any masking; `apply_validity_mask`
then blanked everything else. Clearing `use_coherence_weights` never avoided it, because
`write_velocity_uncertainty` pulls CRLB in regardless.

A pixel with no usable bound now weights **uniformly** and keeps its displacement, velocity,
and temporal coherence. `crlb_sigma`, `velocity_sigma`, and `displacement_variance` still read
as absent there — the fallback recovers the data without manufacturing an uncertainty.

For a single-reference network the SBAS system is exactly determined, so weights cancel and
this is *identical* to the weighted solution. It is only an estimator change for an
over-determined network (`max_bandwidth` set); that is stated in the code and CHANGELOG rather
than hidden.

**Measured on the MMX1/ICMX frame**

| layer | before | after |
|---|---:|---:|
| displacement / velocity / temporal coherence | 0.993062 | **1.000000** |
| crlb_sigma / velocity_sigma / displacement_variance | 0.993062 | 0.993062 |

At the 0.6938% of pixels with no bound: displacement finite in **100%**, temporal coherence
finite in **100%**, `velocity_sigma` NaN in **100%**. Data preserved, bound reported absent.

With #29 the frame goes 98.8613% → 99.3062% → **100%** displacement coverage. Coverage
numbers are unchanged (both stations were already bounded); velocity −251.143 → −251.130 mm/yr
against GNSS −247.811.

Contract: two unit tests in `displacement.rs` —
`a_pixel_with_no_usable_bound_falls_back_to_uniform_weights` and
`an_unbounded_pixel_reports_no_uncertainty`. Verified fmt/clippy clean,
`cargo test --workspace` 261 passed / 0 failed, GNSS harness scores to completion.

Note for #25: this was **not** blocked on the oracle re-pin — `oracle/.venv-v042` exists and
answered the v0.42 question directly.
